hosting/src/models/tanamSite.ts

```diff
@@ -0,0 +1,43 @@
+import { FieldValue, Timestamp, serverTimestamp } from "firebase/firestore";
+
+/**
+ * Tanam site model
+ */
+export class TanamSite {
+  /**
+   *
+   * @param {string} id Site ID
+   * @param {string} title Site title
+   * @param {string} analytics Goolge Analytics tracking ID
+   * @param {Timestamp} createdAt Site creation date
+   * @param Timestamp updatedAt Site last update date
+   */
+  constructor(
+    public readonly id: string,
+    public readonly title: string,
+    public readonly analytics: string,
+    public readonly createdAt: Timestamp,
+    public readonly updatedAt: Timestamp,
+  ) {}
+
+  static fromJson(json: any): TanamSite {
+    return new TanamSite(
+      json.id,
+      json.title,
+      json.analytics,
+      json.created || json.createdAt,
+      json.updated || json.updatedAt,
```

Comment:
```suggestion
     json.createdAt instanceof Timestamp ? json.createdAt : Timestamp.fromDate(new Date(json.createdAt)),
      json.updatedAt instanceof Timestamp ? json.updatedAt : Timestamp.fromDate(new Date(json.updatedAt)),
```

Reply:
1. The JSON can only come from Firestore 🙏 
2. The legacy data is called `created` and `updated`